ratings/views.py

```diff
@@ -24,20 +24,20 @@
 from ratings.models import Rating
 from sounds.models import Sound
 from utils.cache import invalidate_template_cache
+from django.db import transaction
 
 @login_required
+@transaction.atomic()
 def add(request, content_type_id, object_id, rating):
     rating = int(rating)
     content_type = ContentType.objects.get(id=content_type_id)
     if rating in range(1,6):
         # in order to keep the ratings compatible with freesound 1, we multiply by two...
         rating = rating*2
-        try:
-            rating_object = Rating.objects.get(user=request.user, object_id=object_id, content_type=content_type)
-            rating_object.rating = rating;
-            rating_object.save()
-        except Rating.DoesNotExist: #@UndefinedVariable
-            rating_object = Rating.objects.create(user=request.user, object_id=object_id, content_type=content_type, rating=rating)
+        rating_object = Rating.objects.get_or_create(
```

Comment:
`get_or_create` returns `object, created`. If you're not going to use the variable, don't set it